plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4729,6 +4729,17 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
         return VirtualMachineGuestOsIdentifier.OTHER_GUEST;
     }
 
+    private String translateGuestOsIdentifierEx(String cpuArchitecture, String guestOs, String cloudGuestOs) {
+        if (cloudGuestOs != null) {
+            GuestOsDescriptorType guestOsId = GuestOsDescriptorType.fromValue(cloudGuestOs);
+            if (guestOsId != null) {
+                s_logger.debug("Found guest OS mapping name for guest os: " + guestOs);
```

Comment:
@rafaelweingartner the issue is the string concatenation that is performed before the method is invoked.  It is best practice to wrap all debug and trace log calls in ``isDebugEnabled`` or ``isTraceEnabled`` to avoid the pressure on the constant pool and garbage collector from such string concatenation across the codebase.